### PR TITLE
Upgrade libp2p v0.22.0

### PR DESCRIPTION
**Pull request by 0xcb9ff9**
# Description

* upgrade libp2p to v0.22
* fix golang 1.19 build

# Changes include

- [x] Bugfix (non-breaking change that solves an issue)

# Breaking changes

* libp2p compatible across versions 

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have added the relevant labels

## Testing

- [x] I have tested this code with the official test suite

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -24,7 +24,7 @@ build:
 	go build -o dogechain -ldflags="-X 'github.com/dogechain-lab/dogechain/versioning.Version=$(LATEST_VERSION)+$(COMMIT_HASH)+$(DATE)'" main.go
 
 .PHONY: lint
-lint:
+lint: build
```
Comment (DarianShawn) [inline Makefile]: `lint` target should not rely on `build` target.

**Review by DarianShawn — COMMENTED** [review]
Great to see the upgrading. 🙌

We still need some backward compatibility tests though.

File: Makefile (hunk 1 above)
Comment (0xcb9ff9) [reply Makefile]: @DarianShawn  ok, remove it, is my local test command  [e4048e7](https://github.com/dogechain-lab/dogechain/pull/180/commits/e4048e723069beab7022d5e59344f6c3c0153d6c)

**Review by DarianShawn — APPROVED** [review]
LGTM

**Pull request merged**